subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/ConfigurationInternal.java

```diff
@@ -130,18 +130,36 @@ default boolean isDeclarableAgainstByExtension() {
     /**
      * Configures if a configuration can have dependencies declared upon it.
      *
-     * @since 8.0
      */
     void setCanBeDeclaredAgainst(boolean allowed);
 
     /**
      * Returns true if it is allowed to declare dependencies upon this configuration.
      * Defaults to true.
      * @return true if this configuration can have dependencies declared
-     * @since 8.0
      */
     boolean isCanBeDeclaredAgainst();
 
+    boolean isDeprecatedForConsumption();
+    boolean isDeprecatedForResolution();
+    boolean isDeprecatedForDeclarationAgainst();
+
+    void setDeprecatedForConsumption(boolean deprecated);
```

Comment:
Do these need to be part of the API (even internal)? We don't seem to be using them, and theoretically we should be setting these at configuration construction time anyways.

Reply:
See other conversation.